### PR TITLE
Animation Editor: Moved motion event properties to unified inspector window

### PR DESCRIPTION
**Pull request by amzn-jillich**
Moved ownership of the motion event widget to the time view plugin as that is where selection happens and motion event selection is owned.

Signed-off-by: Benjamin Jillich <jillich@amazon.com>

### CODE REVIEW

**Review by galibzon — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 1]
```diff
@@ -14,11 +15,12 @@
...
-#include "../MotionEvents/MotionEventsPlugin.h"
-#include "../MotionEvents/MotionEventPresetsWidget.h"
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionWindowPlugin.h>
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionListWindow.h>
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionSetsWindow/MotionSetsWindowPlugin.h>
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventsPlugin.h>
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventPresetsWidget.h>
+#include <EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.h>
 
 #include "../../../../EMStudioSDK/Source/EMStudioManager.h"
```
Comment (amzn-rhhong) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:25]: Great job for fixing the above line!. Maybe also fix these two line also?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.h [hunk 2]
```diff
@@ -27,11 +27,12 @@ namespace EMStudio
     class MotionEventWidget
         : public QWidget
     {
-        Q_OBJECT
+        Q_OBJECT // AUTOMOC
         MCORE_MEMORYOBJECTCATEGORY(MotionEventWidget, MCore::MCORE_DEFAULT_ALIGNMENT, MEMCATEGORY_STANDARDPLUGINS);
 
     public:
-        MotionEventWidget(QWidget* parent);
+        MotionEventWidget(QWidget* parent = nullptr);
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.h:34]: nit: Might be worth adding `explicit` while we're here to do avoid any strange implicit conversions

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 3]
```diff
@@ -246,6 +254,35 @@ namespace EMStudio
         m_timeViewToolBar->UpdateInterface();
 
         EMotionFX::AnimGraphEditorNotificationBus::Handler::BusConnect();
+
+        // Create the motion event properties widget.
+        m_motionEventWidget = new MotionEventWidget();
+        m_motionEventWidget->hide();
+        connect(this, &TimeViewPlugin::SelectionChanged, this, [=]
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:261]: suggestion/question: Are we sure we want the default capture clause to be by value? Sometimes it's clearer to mention the actual values to be used explicitly to make it clear what is captured. It's a style thing but I usually go for this, up to you on what makes most sense

**Review by hultonha — APPROVED** [review]
LGTM and I think @amzn-rhhong had some good points too worth considering 👍

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp (hunk 1 above)
Comment (amzn-jillich) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:25]: Good call, done

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 4]
```diff
@@ -246,6 +254,35 @@ namespace EMStudio
...
+                {
+                    m_motionEventWidget->ReInit();
+                    m_motionEventWidget->hide();
+                    EMStudio::InspectorRequestBus::Broadcast(&EMStudio::InspectorRequestBus::Events::Clear); // This also gets called when just switching a motion
+                }
+                else
+                {
+                    EventSelectionItem selectionItem = GetSelectedEvent(0);
+                    m_motionEventWidget->ReInit(selectionItem.m_motion, selectionItem.GetMotionEvent());
+                    EMStudio::InspectorRequestBus::Broadcast(&EMStudio::InspectorRequestBus::Events::UpdateWithHeader,
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:279]: Relatively new, came in with the last PR as far as I remember.

We could also replace this with the actual QPixmap, but then the responsibility is for the user to cache that which will result into more duplicated code. Currently, inside the ContentHeaderWidget, we have a hashmap that stores the cached QPixmaps by filename to use the easy interface by filename.

Making it an enumeration would hardcode the inspector to a few given types of content widgets, not a fan of that as the inspector would be flexible and easily extendable in the future without adjusting an enum each time.

What do you think?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.h (hunk 2 above)
Comment (amzn-jillich) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.h:34]: Changed

**Pull request merged**